src/index.ts

```diff
@@ -147,80 +164,106 @@ module.exports = (app: ChartProviderApp): Plugin => {
+
     app.debug(
       `Start charts plugin. Chart paths: ${chartPaths.join(
         ', '
       )}, online charts: ${Object.keys(onlineProviders).length}`
     )
 
-    // Do not register routes if plugin has been started once already
-    pluginStarted === false && registerRoutes()
-    pluginStarted = true
-    const urlBase = `${app.config.ssl ? 'https' : 'http'}://localhost:${
-      'getExternalPort' in app.config ? app.config.getExternalPort() : 3000
-    }`
-    app.debug('**urlBase**', urlBase)
-    app.setPluginStatus('Started')
+    return loadCharts()
+  }
+
+  // Load chart files
+  const loadCharts = async () => {
+    app.debug(`Loading Charts....`)
 
-    const loadProviders = bluebird
-      .mapSeries(chartPaths, (chartPath: string) => findCharts(chartPath))
-      .then((list: ChartProvider[]) =>
-        _.reduce(list, (result, charts) => _.merge({}, result, charts), {})
+    try {
+      const plist = await bluebird.mapSeries(chartPaths, (chartPath: string) =>
+        findCharts(chartPath)
       )
+      const charts = _.reduce(
+        plist,
+        (result, charts) => _.merge({}, result, charts),
+        {}
+      )
+      app.debug(
+        `Chart plugin: Found ${
+          _.keys(charts).length
+        } charts from ${chartPaths.join(', ')}.`
+      )
+      chartProviders = _.merge({}, charts, onlineProviders)
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (e: any) {
+      console.error(`Error loading chart providers`, e.message)
+      chartProviders = {}
+      app.setPluginError(`Error loading chart providers`)
+    }
+  }
 
-    return loadProviders
-      .then((charts: { [key: string]: ChartProvider }) => {
-        app.debug(
-          `Chart plugin: Found ${
-            _.keys(charts).length
-          } charts from ${chartPaths.join(', ')}.`
-        )
-        chartProviders = _.merge({}, charts, onlineProviders)
-      })
-      .catch((e: Error) => {
-        console.error(`Error loading chart providers`, e.message)
-        chartProviders = {}
-        app.setPluginError(`Error loading chart providers`)
-      })
+  const refreshProviders = async () => {
+    const td = Date.now() - (lastWatchEvent as number)
+    app.debug(`last watch event time elapsed = ${td}`)
+    if (lastWatchEvent && td > 5000) {
+      app.debug(`Reloading Charts...`)
+      lastWatchEvent = undefined
+      await loadCharts()
+    }
+  }
+
+  const handleWatchEvent = () => {
+    lastWatchEvent = Date.now()
   }
 
   const registerRoutes = () => {
     app.debug('** Registering API paths **')
 
+    app.debug(`** Registering map tile path (${chartTilesPath} **`)
     app.get(
-      `/signalk/:version(v[1-2])/api/resources/charts/:identifier/:z([0-9]*)/:x([0-9]*)/:y([0-9]*)`,
+      `/${chartTilesPath}/:identifier/:z([0-9]*)/:x([0-9]*)/:y([0-9]*)`,
       async (req: Request, res: Response) => {
         const { identifier, z, x, y } = req.params
+        await refreshProviders()
```

Comment:
To me this style of sprinkling `refreshProviders` seems a bit messy.

I would have implemented this by removing the direct access to `chartProviders`: renaming it to `_chartProviders` or something and introducing `getChartProviders`, where we could have one place to apply the logic of making sure we are serving up to date data.

Reply:
Done.